src/core/expr/fexpr_cumsum.cc

```diff
@@ -23,30 +24,45 @@ class Column_cumsum : public Virtual_ColumnImpl {
       xassert(col_.can_be_read_as<T>());
     }
 
-    void _materialize(Column& out, bool) const {
-      Column newcol = Column::new_data_column(nrows_, stype());
-      T* data = static_cast<T*>(newcol.get_data_editable(0));
-      parallel_for_static(nrows_,
-          [&](size_t i) {
-            data[i] = (i == 0) ? data[i] : data[i] + data[i - 1];
-          });
-        out = newcol;
-      }
-
-      void Column_cumsum::materialize(Column& out, bool) {
-        switch (stype()) {
-          case SType::INT8:    return _materialize<int8_t>(out);
-          case SType::INT16:   return _materialize<int16_t>(out);
-          case SType::INT32:   return _materialize<int32_t>(out);
-          case SType::INT64:   return _materialize<int64_t>(out);
-          case SType::FLOAT32: return _materialize<float>(out);
-          case SType::FLOAT64: return _materialize<double>(out);
-          default:
-            throw RuntimeError() << "Invalid stype for cumsum";
-        }
-      }
+    void materialize(Column& out, bool) override {
+        Column tmp = Column::new_data_column(col_.nrows(), stype());
+        auto data = static_cast<T*>(tmp.get_data_editable());
+         parallel_for_static(col_.nrows(),
+         [&](size_t i) {
+        for (size_t i, i < col_.nrows(), i++) {
```

Comment:
That's not a proper syntax for a for-loop, also, `i` shadows `(size_t i)`. Logic here is also kind of wrong, because `parallel_for_static` already is a loop, so no need to have another loop inside.